### ISSUE TITLE
[feature] User specified variables to templates.

### ISSUE THREAD

**Issue by rasjani** [conan-io/conan#7510]
Conan 1.27 was released with `--template` flag that can be used with to scaffold new conan packages when  used in conjunction with conan new.

Before this, I've  been shipping [cookiecutter](https://github.com/cookiecutter/cookiecutter) templates in my conan config git repository in templates folder that where used to accomplish similar feature - however, cookiecutter has ability to query any values that will then passed to jinja that does the scaffolding. This happens by providing a configuration file to cookiecutter which contains a field name and a default value for that field name. 

For example in my case i have something like this:
```json
{
  "project_name": "project",
  "project_version": "0.0.1",
  "project_description": "No description",
  "url": "https://codereview.xxx.com/plugins/gitiles/Vendor{{cookiecutter.project_name}}/+/refs/heads/master",
  "project_homepage": "",
  "license": "SpdxLicenseId",
  "full_name": "Firstname Lastname",
  "email": "{{cookiecutter.full_name.replace(' ','.').lower()}}@xxx.com",
  "sources_in": "{{cookiecutter.project_name}}-{{cookiecutter.project_version}}",
  "_extensions": ["jinja2_time.TimeExtension"]
}
```

With this, lets say i have a header only package, i do scaffolding which asks me questions in terminal  to fill in these values and only thing i have to do is to drop in the upstream sourcecode into directory i picked up in for `sources_in` variable, commit and it gets build - without 0 modifications to conanfile.py or build.py  after the scaffolding is done unless there's some specific flags needed to be passed to cmake or whatever build system ... 

This approach allows me to populate other sort of fields besides what is now [available](https://docs.conan.io/en/latest/extending/template_system/command_new.html#context)  which makes making new packages a breeze.  I think this would be a better "developer experience" and allows more control in what sort of stuff can go into templates for `conan new`

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
While I think this is a cool feature, I would also prefer to not add yet another dependency (cookiecutter) for this, check the recent discussion and issues here: https://github.com/conan-io/conan/pull/7461, unless it proves very necessary.

I wonder if it would make sense that this is one of the command extensions that are already implemented for Conan 2.0.

Wdyt @jgsogo ?

**Comment by jgsogo** [comment]
Hi, @rasjani , thanks for opening the issue.

I'd love to add more flexibility/power to the `conan new --template`, we just need to find the best way to do it. Probably cookie-cutter itself is a bit overkill and we don't need all its functionality. 

I propose something like:
 * current templates are Jinja2 template, they can define default values ✔️ 
 * we want/need to pass more values using CLI, we can use a `--data=data.json` with key-value pairs:
   ```cmd
   conan new --template=header-only --data=daja.json my_pkg/version
   ```

   but it can accept more variables directly from CLI that will override those in the JSON file (only plain values):

   ```cmd
   conan new --template=header-only --data=daja.json --key:provides=libjpeg my_pkg/version
   ```

 * interactive? I would probably delegate it in a different script, not Conan. You run your script to generate the `data.json` and then you provide that file to `conan new`... or you create your custom Conan command (CLI 2.0) that does the job.

IMO this proposal would be enough for most use-cases and a custom command could cover the rest without adding that complexity to Conan itself.

**Comment by memsharded** [comment]
This was implemented time ago in Conan 2.0, that ``conan new -d var=value`` can inject any arbitrary variables to templates, both for file contents and file names. Closing as solved.
